src/ImageSharp/Common/Helpers/SimdUtils.HwIntrinsics.cs

```diff
@@ -749,17 +752,26 @@ public static ushort TrailingZeroCount(ushort value)
         /// <summary>
         /// <see cref="ByteToNormalizedFloat"/> as many elements as possible, slicing them down (keeping the remainder).
         /// </summary>
+        /// <param name="source">The source buffer.</param>
+        /// <param name="destination">The destination buffer.</param>
         [MethodImpl(InliningOptions.ShortMethod)]
         internal static void ByteToNormalizedFloatReduce(
             ref ReadOnlySpan<byte> source,
-            ref Span<float> dest)
+            ref Span<float> destination)
         {
-            DebugGuard.IsTrue(source.Length == dest.Length, nameof(source), "Input spans must be of same length!");
+            DebugGuard.IsTrue(source.Length == destination.Length, nameof(source), "Input spans must be of same length!");
 
-            if (Avx2.IsSupported || Sse2.IsSupported)
+            if ((Vector512.IsHardwareAccelerated && Avx512F.IsSupported) ||
+                Avx2.IsSupported ||
+                Sse2.IsSupported ||
+                AdvSimd.IsSupported)
             {
                 int remainder;
-                if (Avx2.IsSupported)
+                if (Avx512F.IsSupported)
```

Comment:
```suggestion
                if (Vector512.IsHardwareAccelerated && Avx512F.IsSupported)
```

Reply:
BTW, I'm not 100% sure you want to include the `Vector512.IsHardwareAccelerated` restriction on all of these.  You do have the option of restricting on `IsSupported` only and telling users who don't want to use AVX-512 on the slower hardware to disable it with `DOTNET_EnableAVX512F=0`.  I haven't made this decision for my own libs yet.

Reply:
I think I'll stick and include the check. I want my users to get no surprises.